content/blog/vue-translate.md

```diff
@@ -0,0 +1,175 @@
+ an HIV model developed by researchers at GIDA in association with UNAIDS. To support the West African
+ countries who are the users of Naomi we are providing two language modes - French and English. 
+ 
+ The front-end of the app is written in [Vue.js](https://vuejs.org/) and makes use of [Vuex](https://vuex.vuejs.org/),
+ a [flux](https://facebook.github.io/flux/) library for Vue. There are two ways we could have implemented
+ translations in Vue - with components or directives. We opted for the latter; I'll explain why and how!
+
+### Components vs directives
+Vue, like React, is a framework for writing components. 
+A component looks like a custom HTML element, but has special behaviour attached to it by Vue.
+Vuex provides helper functions that allow
+components to subscribe to changes in the store, so it's pretty straight-forward to define a component that will
+ perform translations. Usage of such a component would look a little like this:
+
+```
+<h3><translated translation-key="welcomeMessage"></translated></h3>
+```
+
+This is a little verbose, but also will become a lot more complicated if we want to translate something 
+other than text, for example the placeholder of an input element. It could certainly be done, but 
+you'd end up with an interface something like:
+
+```
+<translated-input 
+            translation-key="email" 
+            translation-attribute="placeholder" 
+            inputType="text"
+            name="myEmailInput"
+            value="">
+</translated-input>
+```
+
+But unlike React and more like Angular, Vue is also a framework of directives. A directive looks 
+like a custom attribute on an HTML element and defines some behaviour on that element. It would be much
+less intrusive if we could define a custom directive that would translate any part of an existing element.
+The desired interface would be:
+
+```
+<h3 v-translate="'welcomeMessage'"></h3>
+<input type="text" v-translate:placeholder="'email'" name="myEmailInput" value="" />
+```
+
+This is clearly a much nicer interface, but is not as straight-forward to implement. This blog will describe
+how we did it!
+
+### Storing and updating the user's chosen language
+In Vuex all app state is held
+in a central "store", which is then the ideal place to keep track of what language the user has selected.
+
+```
+const store = new Vuex.Store({
+    state: {
+        language: "en     
+    } 
+})
+```
+
+For performing the translations we have used [i18next](https://www.i18next.com/), which is intialised
+ with a language and the available translations. When the user selects a new language we fire off a Vuex
+  action that updates both i18next and the language in our store state.
+ 
+ ```
+ await i18next.changeLanguage(newLanguage);
+ commit({type: "CHANGE_LANGUAGE", payload: newLanguage})
+```
+
+### Listening for changes
+
+Listening for changes in a component is easy, it's baked into the framework. A simple translation component
```

Comment:
Possibly worth adding something for naive people (like me) that this is *not* what you did, as I'd forgotten that we're not doing a component but instead a directive and got confused trying to relate this to your translate.ts implementation.